File: ios/Sources/DevicePlugin/CommunityDevicePlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(CommunityDevicePlugin)
public class CommunityDevicePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "CommunityDevicePlugin"
    public let jsName = "CommunityDevice"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getInfo", returnType: .promise)
    ]
    private let implementation = CommunityDevice()

    @objc func getInfo(_ call: CAPPluginCall) {

        let diskFree = implementation.getFreeDiskSize() ?? 0
        let realDiskFree = implementation.getRealFreeDiskSize() ?? 0
        let diskTotal = implementation.getTotalDiskSize() ?? 0

        call.resolve([
            "diskFree": diskFree,
            "diskTotal": diskTotal,
            "realDiskFree": realDiskFree,
            "realDiskTotal": diskTotal
        ])
    }
}

File: android/src/main/java/com/ryltsov/alex/plugins/device/CommunityDevicePlugin.kt
package com.ryltsov.alex.plugins.device

import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin

@CapacitorPlugin(name = "CommunityDevice")
public class CommunityDevicePlugin : Plugin() {
    private lateinit var implementation: CommunityDevice

    override fun load() {
        implementation = CommunityDevice()
    }

    @PluginMethod
    public fun getInfo(call: PluginCall) {
        val r = JSObject()

        r.put("diskFree", implementation.diskFree)
        r.put("diskTotal", implementation.diskTotal)
        r.put("realDiskFree", implementation.realDiskFree)
        r.put("realDiskTotal", implementation.realDiskTotal)

        call.resolve(r)
    }
}
